src/ucp/wireup/wireup_cm.c

```diff
@@ -275,9 +275,51 @@ ucp_cm_ep_sa_data_pack(ucp_ep_h ep, ucp_wireup_sockaddr_data_base_t *sa_data,
     }
 }
 
+static ucs_status_t
+ucp_cm_ep_priv_space_is_enough(ucp_ep_h ep, const ucp_tl_bitmap_t *tl_bitmap,
+                               ucs_log_level_t log_level,
+                               unsigned ep_init_flags,
+                               ucp_object_version_t sa_data_version)
+{
+    ucp_worker_h worker   = ep->worker;
+    ucp_context_h context = worker->context;
+    unsigned pack_flags   = ucp_cm_address_pack_flags(worker);
+    size_t ucp_addr_size, sa_data_length;
+    ucp_rsc_index_t cm_idx;
+    ucs_status_t status;
+
+    if (ucp_ep_get_cm_wireup_ep(ep)->flags & UCP_WIREUP_EP_FLAG_SEND_CLIENT_ID) {
+        pack_flags |= UCP_ADDRESS_PACK_FLAG_CLIENT_ID;
+    }
+
+    if (ep_init_flags & UCP_EP_INIT_CREATE_AM_LANE_ONLY) {
+        pack_flags |= UCP_ADDRESS_PACK_FLAG_AM_ONLY;
+    }
+
+    status = ucp_address_length(worker, ep, tl_bitmap, pack_flags,
+                                context->config.ext.worker_addr_version,
+                                &ucp_addr_size);
+    if (status != UCS_OK) {
+        return status;
+    }
+
+    sa_data_length = ucp_cm_priv_data_length(ucp_addr_size, sa_data_version);
+    cm_idx         = ep->ext->cm_idx;
+    if (worker->cms[cm_idx].attr.max_conn_priv < sa_data_length) {
+        ucs_log(log_level,
+                "CM private data buffer is too small to pack UCP endpoint "
+                "info, ep %p service data version %u, size %lu, address length "
+                "%lu, cm %p max_conn_priv %lu", ep, sa_data_version,
+                sa_data_length - ucp_addr_size, ucp_addr_size,
+                worker->cms[cm_idx].cm, worker->cms[cm_idx].attr.max_conn_priv);
+        return UCS_ERR_BUFFER_TOO_SMALL;
+    }
```

Comment:
can we unite common code with ucp_cm_ep_priv_data_pack?

Reply:
Yes. I'll do it in next PR update.
Note: Since ```ep``` pointer won't be used as argument in ```ucp_cm_ep_priv_space_is_enough```, after uniting some common code, ```ep``` pointer info won't be reported in ```ucs_log(log_level, ...``` anymore.

Reply:
Done